src/lib/es5.d.ts

```diff
@@ -1747,11 +1743,11 @@ interface Int8Array {
 
     [index: number]: number;
 }
-interface Int8ArrayConstructor {
-    readonly prototype: Int8Array;
-    new(length: number): Int8Array;
-    new(arrayOrArrayBuffer: ArrayLike<number> | ArrayBufferLike): Int8Array;
-    new(buffer: ArrayBufferLike, byteOffset: number, length?: number): Int8Array;
+interface TypedArrayConstructor<T> {

```

Comment:
For now, we should leave the existing typed array constructors as is rather than introduce `TypedArrayConstructor`.

Reply:
Alternatively, we could try the following definition:
```ts
interface TypedArrayConstructor {
    prototype: TypedArray;
    new (length: number): this["prototype"];
    new (arrayOrArrayBuffer: ArrayLike<number> | ArrayBufferLike): this["prototype"];
    new (buffer: ArrayBufferLike, byteOffset: number, length?: number): this["prototype"];
    of(...items: number[]): this["prototype"];
    from(arrayLike: ArrayLike<number>, mapfn?: (v: number, k: number) => number, thisArg?: any): this["prototype"];
}

interface Int8ArrayConstructor extends TypedArrayConstructor {
    prototype: Int8Array;
}
```

Though its possible this could cause problems with `Symbol.species`.